### PR TITLE
Dev dimentions

### PR DESCRIPTION
**Pull request by Valansch**

### CODE REVIEW

File: locale/gen_combined/dimentions.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
+--Author: Valansch
+
+local resource_types = {"copper-ore", "iron-ore", "coal", "stone", "uranium-ore", "crude-oil"}
+
+global.portals = {}
+--Exmaple Portal:
```
Comment (Valansch) [inline locale/gen_combined/dimentions.lua]: Exmaple huehuehue

File: locale/gen_combined/dimentions.lua [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      if distance(portal.position, player.position) < global.portal_radius then
+        if not global.last_tp[player.name] or global.last_tp[player.name] + global.teleport_cooldown * 60 < game.tick then
+          player.teleport(portal.target, portal.target_surface)
+          global.last_tp[player.name] = game.tick
+          player.print("Wooosh! You are now in the " .. get_nice_surface_name(portal.target_surface) .. " dimention.")
+        end
+      end
+    end
+  end
+end
```
Comment (grilledham) [inline locale/gen_combined/dimentions.lua]: This looks fine, I'm just curious if you thought about using surface.find_entities_filtered.

File: locale/gen_combined/dimentions.lua [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        for item, count in pairs(inv.get_contents()) do
+          local n_inserted = target_inv.insert{name = item, count = count}
+          if n_inserted > 0 then
+            inv.remove{name = item, count = n_inserted}
+          end
+        end
+      end
+    end
+    global.current_magic_chest_index = (global.current_magic_chest_index) % num_chests + 1 --Next portal
+  end
```
Comment (grilledham) [inline locale/gen_combined/dimentions.lua]: The comment should probably read Next magic chest.

**Comment by Valansch** [comment]
We have tested an older version of this in MP, its live on the "test bench" server. But not this one. I will test it before merging

File: locale/gen_combined/dimentions.lua (hunk 3 above)
Comment (Valansch) [reply locale/gen_combined/dimentions.lua]: True

File: locale/gen_combined/dimentions.lua (hunk 2 above)
Comment (Valansch) [reply locale/gen_combined/dimentions.lua]: What do you mean? What would i use it for?

Comment (Valansch) [reply locale/gen_combined/dimentions.lua]: In fact i didn't. It's probably faster. I wil change it. Good point.

**Pull request merged**